src/plugins/intel_cpu/src/nodes/executors/acl/acl_fullyconnected.cpp

```diff
@@ -46,104 +52,181 @@ static VectorDims makeDummyOutputDims(const VectorDims& inShape, const VectorDim
-                memoryArgs[ARG_DST] = std::make_shared<Memory>(context->getEngine(),
-                                                               memoryArgs[ARG_SRC_0]->getDescPtr()->clone());
-                auto aclWeightsReorder = std::make_shared<acl_fc_executor::ACLWeightsReorder>(
-                        arm_compute::WeightFormat::OHWI, expectedWeightFormat);
-                if (aclWeightsReorder->update(memoryArgs)) {
-                    aclWeightsReorder->execute(memoryArgs);
-                    final_ptr = memoryArgs[ARG_DST];
-                }
+            if (!reorder) {
+                OPENVINO_THROW("No reorder available for the following tensor descriptors: ",
+                               input->getDesc().serializeFormat(),
+                               " and ",
+                               output->getDesc().serializeFormat());
             }
         }
-        // Transpose weights
+        if (reorder) {
+            dnnl::stream loc_stream(engine, dnnl::stream::flags::in_order);
+            reorder.execute(loc_stream, {{DNNL_ARG_FROM, srcMemory}, {DNNL_ARG_TO, dstMemory}});
+        } else {
+            OPENVINO_THROW("Could not make onednn reorder.");
+        }
+    }
+    return output;
+}
+
+static MemoryPtr prepareWeightMemory(const MemoryArgs &memory,
+                                     const ExecutorContext::CPtr context,
+                                     const FCAttrs &attrs,
+                                     ACLFCAttrs& aclfcAttrs,
+                                     const PostOps &postOps,
+                                     arm_compute::WeightFormat& expectedWeightFormat,
+                                     arm_compute::TensorInfo& wei_tensor_info) {
+    DEBUG_LOG("ACLFullyConnectedExecutor: prepack weights");
+
+    MemoryArgs memoryArgs;
+    memoryArgs[ARG_BIAS]  = memory.at(ARG_BIAS);
+    memoryArgs[ARG_WEI]   = memory.at(ARG_WEI);
+    if (memory.at(ARG_SRC_0)->getShape().isDynamic()) {
+        const auto& inShape = memory.at(ARG_SRC_0)->getShape();
+        const auto& wShape = memory.at(ARG_WEI)->getShape();
+        const auto& inDymmyDims = makeDummyInputDims(inShape, wShape);
+        const auto& outDymmyDims = makeDummyOutputDims(inDymmyDims, wShape.getStaticDims(), memory.at(ARG_DST)->getShape().getRank());
+        memoryArgs[ARG_SRC_0] = std::make_shared<Memory>(context->getEngine(),
+                                                            memory.at(ARG_SRC_0)->getDescPtr()->cloneWithNewDims(inDymmyDims));
+        memoryArgs[ARG_DST] = std::make_shared<Memory>(context->getEngine(),
+                                                        memory.at(ARG_DST)->getDescPtr()->cloneWithNewDims(outDymmyDims));
+    } else {
+        memoryArgs[ARG_SRC_0] = memory.at(ARG_SRC_0);
+        memoryArgs[ARG_DST]   = memory.at(ARG_DST);
+    }
+    auto aclWeightsRepack = std::make_shared<acl_fc_executor::ACLWeightFormatGenerator>(attrs, postOps, memoryArgs);
```

Comment:
I think in a long term we should drop ACLWeightFormatGenerator and just use a future reorderExecutor.
So, in case oneDNN is not available, reorderExecutor will call NEReorder+NETranspose.
If oneDNN is available - call dnnl::reorder().
Could you please leave a comment.

Reply:
Added